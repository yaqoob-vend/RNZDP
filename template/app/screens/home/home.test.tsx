import {HomeScreen} from '.';
import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import {BottmTabNavigatorParamList} from '../../navigators';
import {BottomTabNavigationProp} from '@react-navigation/bottom-tabs';
import {SafeAreaProvider} from 'react-native-safe-area-context';

type HomeScreenProps = BottomTabNavigationProp<
  BottmTabNavigatorParamList,
  'Home'
>;
describe('Logout & navigate to login screen', () => {
  let navigation: Partial<HomeScreenProps>;
  beforeEach(() => {
    navigation = {
      navigate: jest.fn(),
    };
  });
  test('successfully logout', async () => {
    const {getByText, toJSON} = render(
      <SafeAreaProvider
        initialSafeAreaInsets={{top: 1, left: 2, right: 3, bottom: 4}}>
        <HomeScreen navigation={navigation as HomeScreenProps} route={null} />,
      </SafeAreaProvider>,
    );
    const button = getByText('Logout');
    fireEvent.press(button);

    waitFor(() => expect(navigation.navigate).toBeCalled());
    expect(toJSON()).toMatchSnapshot();
  });
});
