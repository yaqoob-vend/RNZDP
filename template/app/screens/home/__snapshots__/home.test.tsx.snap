// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout & navigate to login screen successfully logout 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="HomeScreen"
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "flex": 1,
              "height": "100%",
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "flex": 1,
              "height": "100%",
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
            },
            Object {
              "paddingTop": 1,
            },
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {
                "alignItems": "stretch",
                "justifyContent": "flex-start",
              },
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "paddingHorizontal": 16,
              },
            ]
          }
          keyboardShouldPersistTaps="handled"
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "height": "100%",
              },
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0)",
              },
            ]
          }
        >
          <View>
            <RCTImageView
              internal_analyticTag={null}
              resizeMode="cover"
              source={
                Array [
                  Object {
                    "height": undefined,
                    "uri": undefined,
                    "width": undefined,
                  },
                ]
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 100,
                  "overflow": "hidden",
                  "width": 100,
                }
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Helvetica",
                    "fontSize": 24,
                    "fontWeight": "bold",
                  },
                  Object {
                    "color": "black",
                    "fontFamily": "Helvetica",
                    "fontSize": 28,
                    "fontWeight": "bold",
                    "lineHeight": 38,
                    "textAlign": "center",
                  },
                ]
              }
            >
              welcomeScreen.readyForLaunch.test
            </Text>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FBA928",
                  "borderRadius": 4,
                  "justifyContent": "center",
                  "marginHorizontal": 10,
                  "opacity": 1,
                  "paddingHorizontal": 8,
                  "paddingVertical": 8,
                }
              }
              testID="Logout"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Helvetica",
                      "fontSize": 15,
                    },
                    Array [
                      Object {
                        "color": "#ffffff",
                        "fontSize": 9,
                        "paddingHorizontal": 12,
                      },
                      Object {
                        "fontSize": 15,
                      },
                    ],
                  ]
                }
              >
                Logout
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
  ,
</RNCSafeAreaProvider>
`;
